spec/services/courses/query_spec.rb

```diff
@@ -307,6 +307,82 @@ def initialize(course, minimum_distance_to_search_location:)
       end
     end
 
+    context 'when searching by provider' do
```

Comment:
I think this is missing the logic of the current recruitment cycle. When searching by ID, you don't allow it to match with providers from another cycle. We should test this.

Reply:
I agree. That's done.